### PR TITLE
Fix Packagist JSON parsing to correctly parse dist fields.

### PR DESCRIPTION
**Pull request by calebbrown**
Also handle the scenario where a download url is not available.

### CODE REVIEW

File: internal/pkgmanager/ecosystem.go [hunk 1]
```diff
@@ -9,6 +10,8 @@ import (
 	"github.com/ossf/package-analysis/pkg/api/pkgecosystem"
 )
 
+var ErrNoArchiveURL = errors.New("no archive url found")
```
Comment (maxfisher-g) [inline internal/pkgmanager/ecosystem.go:13]: nit: `"archive URL not found"` sounds a bit more natural

File: internal/pkgmanager/packagist.go [hunk 2]
```diff
@@ -10,19 +10,31 @@ import (
...
+	Type      string `json:"type"`
+	Shasum    string `json:"shasum,omitempty"`
+	Reference string `json:"reference"`
+}
+
+func (d *packagistDistJSON) UnmarshalJSON(data []byte) error {
+	switch string(data) {
+	case "null":
+		return nil
+	case "\"__unset\"":
```
Comment (maxfisher-g) [inline internal/pkgmanager/packagist.go:24]: nit: use backticks for quotation

File: internal/pkgmanager/packagist.go [hunk 3]
```diff
@@ -10,19 +10,31 @@ import (
...
+		return nil
+	}
+	return json.Unmarshal(data, &d)
```
Comment (maxfisher-g) [inline internal/pkgmanager/packagist.go:27]: nit: infinite loop

**Review by maxfisher-g — CHANGES_REQUESTED** [review]
some nits, including an infinite loop

File: internal/pkgmanager/packagist.go (hunk 3 above)
Comment (calebbrown) [reply internal/pkgmanager/packagist.go]: fixed

File: internal/pkgmanager/packagist.go (hunk 2 above)
Comment (calebbrown) [reply internal/pkgmanager/packagist.go]: fixed

File: internal/pkgmanager/ecosystem.go (hunk 1 above)
Comment (calebbrown) [reply internal/pkgmanager/ecosystem.go]: fixed

**Review by maxfisher-g — APPROVED** [review]

**Pull request merged**